polkadot/runtime/parachains/src/runtime_api_impl/v5.rs

```diff
@@ -52,29 +52,15 @@ pub fn validator_groups<T: initializer::Config>(
 /// Implementation for the `availability_cores` function of the runtime API.
 pub fn availability_cores<T: initializer::Config>() -> Vec<CoreState<T::Hash, BlockNumberFor<T>>> {
 	let cores = <scheduler::Pallet<T>>::availability_cores();
-	let config = <configuration::Pallet<T>>::config();
 	let now = <frame_system::Pallet<T>>::block_number() + One::one();
-	let rotation_info = <scheduler::Pallet<T>>::group_rotation_info(now);
 
-	let time_out_at = |backed_in_number, availability_period| {
```

Comment:
Don't duplicate time out logic, but use same function as the scheduler uses to avoid this to diverge from reality.